src/Symfony/Component/Routing/Router.php

```diff
@@ -271,9 +271,9 @@ public function matchRequest(Request $request)
     }
 
     /**
-     * Gets the UrlMatcher instance associated with this Router.
+     * Gets the UrlMatcher or RequestMatcher instance associated with this Router.
      *
-     * @return UrlMatcherInterface A UrlMatcherInterface instance
+     * @return UrlMatcherInterface|RequestMatcherInterface
```

Comment:
If you scroll a few lines up, you can see that this is actually the expected behavior of the method.